hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientInvocationServiceImpl.java

```diff
@@ -106,20 +107,17 @@ protected void send(ClientInvocation invocation, ClientConnection connection) th
         registerInvocation(invocation);
 
         ClientMessage clientMessage = invocation.getClientMessage();
-        if (!isAllowedToSendRequest(connection, invocation) || !writeToConnection(connection, clientMessage)) {
-            final long callId = clientMessage.getCorrelationId();
-            ClientInvocation clientInvocation = deRegisterCallId(callId);
-            if (clientInvocation != null) {
-                throw new IOException("Packet not send to " + connection.getEndPoint());
-            } else {
-                if (invocationLogger.isFinestEnabled()) {
-                    invocationLogger.finest("Invocation not found to deregister for call ID " + callId);
-                }
-                return;
-            }
+        if (isAllowedToSendRequest(connection, invocation) && writeToConnection(connection, clientMessage)) {
+            invocation.setSendConnection(connection);
+            return;
         }
 
-        invocation.setSendConnection(connection);
+        long callId = clientMessage.getCorrelationId();
+        ClientInvocation clientInvocation = deRegisterCallId(callId);
+        if (clientInvocation != null) {
+            throw new WrongTargetException("Could not found member " + connection.getEndPoint()
```

Comment:
On the server we throw WrongTargetException; but it isn't the most informing exception. TargetDisconnectedException is imho easier to digest... but then client and server will not be consistent.